### PR TITLE
Added Long Word List Functionality

### LINKED ISSUE #3: Generate List of Long Words
Program should generate a text file containing all the words in input with more than 7 letters in them.
File should be stored "in the current directory".

### PR DESCRIPTION
**Pull request by Stevie-Goodman**
Resolves #3

### CODE REVIEW

File: Analyse.cs [hunk 1]
```diff
@@ -53,4 +59,21 @@ public Analyse(Input input)
...
+    /// <param name="input">The Input object to review.</param>
+    /// <returns>An array of words longer than 7 letters.</returns>
+    private List<string> GetLongWords(Input input)
+    {
+        List<string> longWords = 
+            (
+            from word in Regex.Matches(input.InputText, "[a-z']{8,}")
+            where word.Value.Length > 7
+            select word.Value
+            ).ToList(); // Converts from IEnumerable<string> to List<string>.
```
Comment (Stevie-Goodman) [inline Analyse.cs:75]: Redundant `where` statement on line 73.
```suggestion
        List<string> longWords = 
            (
            from word in Regex.Matches(input.InputText, "[a-z']{8,}")
            select word.Value
            ).ToList(); // Converts from IEnumerable<string> to List<string>.
```

File: Analyse.cs [hunk 2]
```diff
@@ -53,4 +59,20 @@ public Analyse(Input input)
...
+    /// <summary>
+    /// Finds all words longer than 7 letters.
+    /// </summary>
+    /// <param name="input">The Input object to review.</param>
+    /// <returns>An array of words longer than 7 letters.</returns>
+    private List<string> GetLongWords(Input input)
+    {
+        List<string> longWords = 
+            (
+            from word in Regex.Matches(input.InputText, "[a-z']{8,}")
```
Comment (binaryoverload) [inline Analyse.cs:72]: Might it be worth using `[A-Za-z]` to also capture uppercase letters?

You could also use `\w` which will capture all "word" characters

**Review by binaryoverload — CHANGES_REQUESTED** [review]
Looks good! Just a comment regarding the regex 😄

File: Analyse.cs (hunk 2 above)
Comment (Stevie-Goodman) [reply Analyse.cs:72]: Good catch! I'll add an ignore case option.
Thanks.
```suggestion
            from word in Regex.Matches(input.InputText, "[\w']{8,}", RegexOptions.IgnoreCase)
```

File: Analyse.cs [hunk 3]
```diff
@@ -55,5 +61,20 @@ public Analyse(Input input)
...
+    /// <summary>
+    /// Finds all words longer than 7 letters.
+    /// </summary>
+    /// <param name="input">The Input object to review.</param>
+    /// <returns>An array of words longer than 7 letters.</returns>
+    private List<string> GetLongWords(Input input)
+    {
+        List<string> longWords = 
+            (
+            from word in Regex.Matches(input.InputText, "[\w']{8,}", RegexOptions.IgnoreCase)
```
Comment (Stevie-Goodman) [inline Analyse.cs:74]: Remove redundant RegexOptions argument.
```suggestion
            from word in Regex.Matches(input.InputText, "[\w']{8,}")
```

**Pull request merged**